test/suites/scale/deprovisioning_test.go

```diff
@@ -0,0 +1,630 @@
+				env.EventuallyExpectInitializedNodeCount("==", expectedNodeCount)
+				env.EventuallyExpectHealthyPodCount(selector, replicas)
+			}, aws.ProvisioningEventType, emptinessTestGroup, defaultTestName)
+
+			env.Monitor.Reset() // Reset the monitor so that we now track the nodes starting at this point in time
+
+			By("waiting for all deployment pods to be deleted")
+			// Fully scale down all pods to make nodes empty
+			deployment.Spec.Replicas = lo.ToPtr[int32](0)
+			env.ExpectDeleted(deployment)
+			env.EventuallyExpectHealthyPodCount(selector, 0)
+
+			env.MeasureDurationFor(func() {
+				By("kicking off deprovisioning emptiness by setting the ttlSecondsAfterEmpty value on the provisioner")
+				provisioner.Spec.TTLSecondsAfterEmpty = lo.ToPtr[int64](0)
+				env.ExpectCreatedOrUpdated(provisioner)
+
+				env.EventuallyExpectDeletedNodeCount("==", expectedNodeCount)
+				env.EventuallyExpectNodeCount("==", 0)
+				env.EventuallyExpectHealthyPodCount(selector, replicas)
+			}, aws.DeprovisioningEventType, emptinessTestGroup, defaultTestName)
+		}, SpecTimeout(10*time.Minute))
+	})
+	Context("Expiration", func() {
+		It("should expire all nodes", func(_ context.Context) {
+			replicasPerNode := 20
+			maxPodDensity := replicasPerNode + dsCount
+			expectedNodeCount := 20 // we're currently doing around 1 node/2 mins so this test should run deprovisioning in about 45m
+			replicas := replicasPerNode * expectedNodeCount
+
+			deployment.Spec.Replicas = lo.ToPtr[int32](int32(replicas))
+			provisioner.Spec.KubeletConfiguration = &v1alpha5.KubeletConfiguration{
+				MaxPods: lo.ToPtr[int32](int32(maxPodDensity)),
+			}
+
+			By("waiting for the deployment to deploy all of its pods")
+			env.ExpectCreated(deployment)
+			env.EventuallyExpectPendingPodCount(selector, replicas)
+
+			env.MeasureDurationFor(func() {
+				By("kicking off provisioning by applying the provisioner and nodeTemplate")
+				env.ExpectCreated(provisioner, nodeTemplate)
+
+				env.EventuallyExpectCreatedMachineCount("==", expectedNodeCount)
+				env.EventuallyExpectCreatedNodeCount("==", expectedNodeCount)
+				env.EventuallyExpectInitializedNodeCount("==", expectedNodeCount)
+				env.EventuallyExpectHealthyPodCount(selector, replicas)
+			}, aws.ProvisioningEventType, expirationTestGroup, defaultTestName)
+
+			env.Monitor.Reset() // Reset the monitor so that we now track the nodes starting at this point in time
+
+			env.MeasureDurationFor(func() {
+				By("kicking off deprovisioning expiration by setting the ttlSecondsUntilExpired value on the provisioner")
+				// Change Provisioner limits so that replacement nodes will use another provisioner.
+				provisioner.Spec.Limits = disableProvisioningLimits
+				// Enable Expiration
+				provisioner.Spec.TTLSecondsUntilExpired = lo.ToPtr[int64](0)
+
+				noExpireProvisioner := test.Provisioner(test.ProvisionerOptions{
+					Requirements: []v1.NodeSelectorRequirement{
```

Comment:
Consider just copying the requirements from the base provisioner here.
```suggestion
					Requirements: provisioner.Spec.Requirements
```